topics/variant-analysis/tutorials/trio-analysis/tutorial.md

```diff
@@ -0,0 +1,429 @@
+
+> ### {% icon hands_on %} Hands-on: Convert compressed vcf to uncompressed.
+>
+> Open the collection of VCFs and execute the following steps for each VCF.
+>
+> 1. Click on the **pencil** {% icon galaxy-pencil %} of the vcf you want to convert.
+>
+> 2. Click on the **convert tab** {% icon galaxy-gear %}.
+>
+> 3. Under **Target datatype** select `vcf (using 'Convert compressed file to uncompressed.')`
+>
+> 4. Click {% icon exchange %} **Create Dataset**.
+>
+> After transforming all the VCFs you need to combine the converted VCFs into a colllection again.
+>
+> > ### {% icon tip %} Tip: Adding files to collection
+> >
+> > 1. Click on the dataset **selector icon** {% icon param-check %} in your history.
+> > 2. Select the 3 vcf files.
+> > 3. Click on the button **For all selected...** and select **Build Dataset List**
+> > 4. Give the new collection a name and click on **Create collection**.
+> >
+> {: .tip}
+>
+{: .hands_on}
+
+## Add chromosome prefix to vcf
+Again, our next tool has some assumptions about our input VCFs. This time the tool expects the chromosome numbers to start with a prefix `chr`. Our VCFs only use the chromosome numbers however, the VCFs just use the chromosome numbers. This is due to a difference in reference genome used when creating the VCFs. To change the prefix in the VCFs we will use regex again.
+
+> ### {% icon hands_on %} Hands-on: Add chr prefix using regex
+>
+> 1. {% tool [Column Regex Find And Replace](toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.1) %} with the following parameters:
+>    - {% icon param-file %} *"Select cells from"*: `Decompressed VCFs` (output of **Convert compressed file to uncompressed.** {% icon tool %})
+>    - *"using column"*: `Column: 1`
+>    - In *"Check"*:
+>        - {% icon param-repeat %} *"Insert Check"*
+>            - *"Find Regex"*: `(?=^[0-9MYX])`
+>            - *"Replacement"*: `chr`
+>
+> > ### {% icon comment %} Comment: Explaining the regex.
+> >
+> > The regex pattern might look complicated but it's quite simple if you break it down in components.
+> >  - The first component is a positive lookahead `q(?=p)`. Here q is the matched part and p is the pattern. So what it does is to match everything (q) that comes before pattern (p).
+> > - The pattern here is `^[0-9MYX]`, which can again be broken down into different components
+> >      - `^` means that the pattern has to start at the beginning of the line not somewhere randomly in the line. This way our prefix `chr` will be inserted at the start of the line.
+> >      - `[0-9MYX]` means that at this position there should be a character from the list `[]`, namely either a number from `0-9` or the character `M`, `Y`, or `X`.

```

Comment:
Quite a beast of a regex!
A much simpler solution is: `^([0-9MYX])` as the pattern to match and `chr\1` as the replacement pattern.

Reply:
If you want to teach a bit more then consider adding a second "Check" with:
`^(##contig=<.*ID=)([0-9MYX].+)` as the pattern to match and `\1chr\2` as the replacement pattern, which will keep the header consistent.

Reply:
@hexylena any preference here regarding the regex tool? The GalaxyP one used here does the job and so would tool_id=toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/1.1.3

Reply:
The galaxyp one is less likely to be installed everywhere (but ok so is ega download tool). I think I have a mild preference for the `tp_replace_in_column` here as it's more likely to be available, sorry for the extra steps @JasperO98.

And agreed regarding the need for the lookahead, if seems like it'd be functional without it since there's no `q` per your description.

Reply:
Hmm, I didn't know you could do that with regex. I've replaced the prefix check and added your check for the header lines. Thanks! 0816782f5e10f9b06a6fd96dccc51acca59bd54f